src/parallel/HighsTaskExecutor.h

```diff
@@ -138,34 +171,21 @@ class HighsTaskExecutor {
 
   static void initialize(int numThreads) {
     auto& executorHandle = threadLocalExecutorHandle();
-    if (!executorHandle.ptr) {
-      executorHandle.ptr =
-          cache_aligned::make_shared<HighsTaskExecutor>(numThreads);
-      executorHandle.ptr->mainWorkerHandle.store(&executorHandle,
-                                                 std::memory_order_release);
+    if (executorHandle.ptr == nullptr) {
+      executorHandle.ptr = new (cache_aligned::alloc(sizeof(HighsTaskExecutor)))
+          HighsTaskExecutor(numThreads);
+      executorHandle.isMain = true;
```

Comment:
This is absolutely fine, but I'd feel more comfortable if `isMain = true` was set before `ptr` was constructed.